src/common/GroupAvatar.js

```diff
@@ -18,13 +17,24 @@ const styles = StyleSheet.create({
 });
 
 type Props = $ReadOnly<{|
-  names: string,
+  names: string[],
   size: number,
   shape: 'rounded' | 'square',
   children?: React$Node,
   onPress?: () => void,
 |}>;
 
+export const initialsForGroupIcon = (names: string[]): string => {
+  const initialsInCodePoint = names.map(item => item.codePointAt(0));
+  if (initialsInCodePoint.length > 4) {
+    initialsInCodePoint[3] = 8230; // Decimal CodePoint of '…' is 8230
+  }
```

Comment:
This is still, as I noted earlier, "trying to work with data that's partly in one world and partly in another."

I'm not seeing a good reason not to put the `String.fromCodePoint(...)` call in the first `.map`.